### PR TITLE
Create back port of 3.0.0 CSB schema

### LINKED ISSUE #4: Create "back-port" 3.0.0 schema
Create "back-port" CSB 3.0.0 schema that validates provisional B12 v. 3.0.0 JSON encodings specified here:

- https://www.ngdc.noaa.gov/iho/SampleCSBFileFormats.pdf
- https://www.ngdc.noaa.gov/iho/example_csb_geojson_file.geojson

### PR DESCRIPTION
**Pull request by selimnairb**
Resolves issue #4, providing a "back-port" CSB 3.0.0 schema that validates provisional B12 v. 3.0.0 JSON encodings. See #4 for more info.

### CODE REVIEW

**Comment by selimnairb** [comment]
@georgiannazelenak This branch contains schemas that validate the 3.0.0 examples here:

- https://www.ngdc.noaa.gov/iho/SampleCSBFileFormats.pdf
- https://www.ngdc.noaa.gov/iho/example_csb_geojson_file.geojson

File: docs/IHO/b12_v3_2_0-BETA_example.json [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        "soundSpeedDocumented": true,
+        "positionOffsetsDocumented": true,
+        "dataProcessed": true,
+        "contributorComments": "On 2022-03-08, at 20:30 UTC, the echo sounder lost bottom tracking after the vessel crossed another vessel's wake."
+	    },
+      "processing": [
+        {
+          "type": "soundSpeed",
+          "timestamp": "2023-02-14T01:00:00.0000Z",
+          "method": "Interpolated RMC Messages",
```
Comment (brian-r-calder) [inline docs/IHO/b12_v3_2_0-BETA_example.json:60]: Not an issue for the code, but this wouldn't be a method to generate sound speed corrections.
```suggestion
          "method": "WOA18 Profile",
```

**Review by brian-r-calder — APPROVED** [review]

File: tests/unit/test_validators.py [hunk 2]
```diff
@@ -4,53 +4,118 @@
...
+        (valid, result) = validate_b12_3_1_0_2023_03(b12_filepath)
         self.assertTrue(valid)
         self.assertTrue(isinstance(result, dict))
         document: dict = result['document']
         self.assertTrue(isinstance(document, dict))
         with self.assertRaises(KeyError):
-            errors: dict = result['errors']
+            _: dict = result['errors']
+
+        # Validate a valid file without processing metadta
```
Comment (georgiannazelenak) [inline tests/unit/test_validators.py:60]: Check metadata spelling

File: docs/IHO/b12_v3_2_0-BETA_example.json (hunk 1 above)
Comment (selimnairb) [reply docs/IHO/b12_v3_2_0-BETA_example.json]: This has been resolved in a049f3ff57ca6c5cd27233d9fbdf264c2a8033a7.

File: tests/unit/test_validators.py (hunk 2 above)
Comment (selimnairb) [reply tests/unit/test_validators.py]: This has been resolved in aa9d10a9afb41a06419c075d377c31b0323791ee.

**Comment by selimnairb** [comment]
@brian-r-calder @georgiannazelenak I've resolved the outstanding issues. Let me know if this is ready to merge.

**Comment by georgiannazelenak** [comment]
I am still in the process of reviewing, I was only able to get through a portion of the documents on Friday.

File: csbschema/data/XYZ-CSB-schema-3_0_0-2023-03.json [hunk 3]
```diff
@@ -0,0 +1,576 @@
...
+      ],
+      "description": "This field describes the format and version for the data and metadata, such as GeoJSON, CSV, or XYZT. Reference the version of the CSB data convention (e.g., CSB 2.0, CSB 3.0) where possible."
+    },
+    "DataLicense": {
+      "type": "string",
+      "description": "The license under which the Trusted Node is\n providing CSB data to the IHO DCDB.\n"
+    },
+    "UniqueVesselID": {
+      "type": "string",
+      "pattern": "^[a-zA-Z][a-zA-Z0-9]*-[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$",
```
[the remark this answers is not in the document]
Comment (selimnairb) [reply csbschema/data/XYZ-CSB-schema-3_0_0-2023-03.json:115]: @georgiannazelenak Thank you very much for this. This has been addressed in 2a8fbce87383177138d1ecaab35a7023181b18e7.

File: csbschema/data/CSB-schema-3_0_0-2023-03.json [hunk 4]
```diff
@@ -0,0 +1,673 @@
...
+      ],
+      "description": "This field describes the format and version for the data and metadata, such as GeoJSON, CSV, or XYZT. Reference the version of the CSB data convention (e.g., CSB 2.0, CSB 3.0) where possible."
+    },
+    "DataLicense": {
+      "type": "string",
+      "description": "The license under which the Trusted Node is\n providing CSB data to the IHO DCDB.\n"
+    },
+    "UniqueVesselID": {
+      "type": "string",
+      "pattern": "^[a-zA-Z][a-zA-Z0-9]*-[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$",
```
[the remark this answers is not in the document]
Comment (selimnairb) [reply csbschema/data/CSB-schema-3_0_0-2023-03.json]: @georgiannazelenak Thank you very much for this. This has been addressed in 2a8fbce87383177138d1ecaab35a7023181b18e7.

**Comment by selimnairb** [comment]
@brian-r-calder I have addressed the comments of @georgiannazelenak so I think this is ready to be merged (after #2 has been merged).

**Review by georgiannazelenak — APPROVED** [review]

**Pull request merged**